indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,359 @@
+                        justUse = $('<div>').attr({
+                                'class': 'just-use-option'
+                            })
+                            .on('click', function() {
+                                node.trigger('indico:click-on-custom-option', {item_name: node.val()});
+                            });
+                        resultsWrapper.append(justUse);
+                    }
+                    justUse.html("Just use &quot;" + node.val() + "&quot;")
+                } else if (justUse.length != 0) {
+                    justUse.remove();
+                }
+                return resultHtmlList;
+            }
+
+            function buildBackdrop() {
+                var wrapper = $('.dropback-wrapper');
+                var wrapperCloseButton = latestUsedField.parent().find('.close-backdrop-button');
+                if (wrapper.length === 0) {
+                    var dropbackWrapper = $('<div>').attr({
+                        'class': 'dropback-wrapper'
+                    }).on('click keydown', function(e) {
+                        if (latestUsedField.val() != "") {
+                            e.stopPropagation();
+                        } else {
+                            $(this).hide();
+                            latestUsedField.parent().find('.close-backdrop-button').hide();
+                        }
+                    });
+                    var closeButton = $('<a>').attr({
+                        'class': 'icon-close close-button'
+                    }).on('click', function() {
+                        $(this).parent().hide();
+                    });
+                    dropbackWrapper.append(closeButton);
+                    latestUsedField.closest('.i-location-field').append(dropbackWrapper);
+                    wrapperCloseButton.show();
+                } else {
+                    wrapper.show();
+                    wrapperCloseButton.show();
+                }
+            }
+
+            function postSelectionActions(inputField, value) {
+                inputField.val(value);
+                field.val(JSON.stringify(hiddenData));
+                inputField.trigger('indico:close-results-list');
+            }
+
+            function highlightOption(inputField, value) {
+                var highlighter = inputField.parent().find('.highlighter');
+                highlighter.html(value).addClass('visible');
+            }
+
+            function cleanQuery(query) {
+                if (query) {
+                    query = query.trim();
+                    if (!query) {
+                        query = ''
+                    }
+                }
+                return query
+            }
+
+            usingDefault.on('click', function() {
+                toggleDefaultLocation($(this).prop('checked'));
+            });
+
+            toggleDefaultLocation(usingDefault.prop('checked'));
+
+            venueInput.typeahead({
```

Comment:
does this return the typeahead widget? in that case i'd rather assign it instead of using the global `Typeahead[..]`